### PR TITLE
Fixed bug pvd_record #3259

### PR DESCRIPTION
**Pull request by sairajat**
Fixes a bug in write_pvd_record as mentioned in #3259

### CODE REVIEW

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -6220,7 +6220,8 @@ write_pvd_record (std::ostream &out,
                   const std::vector<std::pair<double,std::string> >  &times_and_names) const
 {
   AssertThrow (out, ExcIO());
-
+   
+  out.precision(12);
```
Comment (bangerth) [inline source/base/data_out_base.cc]: In ASPECT, we actually did something recently: unconditionally setting the precision in which we output the time to 12 digits or so. I think it makes sense -- it's just a single number, so the amount of data we write is not of any kind of relevance, and it's a hassle to have to do so through a parameter. I'm in favor of unconditionally leaving it at 12 digits.

Comment (davydden) [inline source/base/data_out_base.cc]: at least `out` stream should be set to the previous precision.

Comment (bangerth) [inline source/base/data_out_base.cc]: Yes, that is correct.

Comment (jppelteret) [inline source/base/data_out_base.cc]: Fair enough, I'm happy with that if we're going to consistently use it from this point forward :-)

Comment (sairajat) [inline source/base/data_out_base.cc]: Yes, it certainly doesn't hurt to set it to 15 as well. Its just a number so, the amount of data is irrelevant. 

Yes, it makes more sense to set the precision to its original state.

Should I set it to 12 or 15 then ?

Comment (bangerth) [inline source/base/data_out_base.cc]: So here's my thought about this: You need higher precision if you have simulations that use time steps that are very different, or if you have many time steps. For example, assume you do 10^7 time steps of size 1. In that case, you lose information when you add up the current time step to the current time, and output it with a small number of digits. In my example, the printed time only changes once every 10 time steps. This leads to all sorts of problems when visualizing things. 

You can avoid that if you output with 8 or 10 or 12 digits. I think 15 is not necessary, though, because I think it is unlikely that you will do 10^15 time steps, and furthermore in that case you get into trouble because *adding time step to time* already incurs round-off on the order of the 15th digit. In other words, the issue then is no longer that we throw away information by printing with too few digits, but that we print information that is no longer correct in the digits that we print. 

In other words, I think 12 is totally sufficient. I can't see a good reason why we *should* print more digits, though of course it also doesn't hurt. (I would just go with 12 consequently.)

**Comment by bangerth** [comment]
Apparently superseded by #3308.

@sairajat  -- There is actually an easier way to achieve what you just did: you can just update your local branch with the changes you made in response to review, and then push this local branch to github. github will then automatically update the pull request with your newest changes!

**Comment by sairajat** [comment]
@bangerth Thank you professor. 
I didn't know that. Also, when I thought about that, my branch was 30 commits behind dealii::master. (It displayed on my github page). So, I was not sure how to update my branch.

Hence, I updated my master, deleted old branch, then created a new branch and created a pull request.

I am sure there are better ways to do this. I think you should have more video-lectures on this topic :). I took help from 32.8 lecture and google :)

**Comment by bangerth** [comment]
It doesn't matter that you are 30 commits behind. With as much development as we do, it is not possible to always be on the most recent patch. The solution is to let `git` and github handle this, which they happily do.

**Comment by bangerth** [comment]
As for videos: I think there are already many resources on the web about using github, and many of them are much better than I could possibly record myself :-)
